R/deprecate.R

```diff
@@ -362,6 +362,33 @@ loadFromHumann <- function(...) {
     importHUMAnN(...)
 }
 
+#' @rdname deprecate
+#' @export 
+setGeneric("splitByRanks",
+           signature = "x",
+           function(x, ...)
+               standardGeneric("splitByRanks"))
+
+#' @rdname deprecate
+#' @export 
+setMethod("splitByRanks", signature = c(x = "SummarizedExperiment"),
+            function(x,...){
+                .Deprecated(msg = paste0("'splitByRanks' is deprecated. ",
+                                        "Use 'agglomerateByRanks' instead."))
+                agglomerateByRanks(x, as.list = FALSE,...)
+            }
+)
+
+#' @rdname deprecate
+#' @export 
+setMethod("splitByRanks", signature = c(x = "SingleCellExperiment"),
+          function(x,...){
+              .Deprecated(msg = paste0("'splitByRanks' is deprecated. ",
+                                       "Use 'agglomerateByRanks' instead."))
+              agglomerateByRanks(x, as.list = FALSE,...)
+          }
+)
+
```

Comment:
After considering this, you could leave splitByRanks as an alias.

1. SingleCellExperiment has fnction splitAltExps which is simialr to this
2. We have still unsplitByRanks

--> Add splitByRanks function back, but like it is here: just empty function calling agglomerateByRanks. (Also, only SummarizedExperiment method is needed since SCE is SE)